cmd/file_lint.go

```diff
@@ -0,0 +1,165 @@
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"strings"
+
+	"github.com/daveshanley/vacuum/motor"
+	"github.com/daveshanley/vacuum/rulesets"
+	"github.com/kong/go-apiops/filebasics"
+	"github.com/kong/go-apiops/logbasics"
+	"github.com/spf13/cobra"
+)
+
+var (
+	cmdLintInputFilename string
+	cmdLintInputRuleset  string
+)
+
+func executeLint(cmd *cobra.Command, _ []string) error {
+	severityList := map[string]int{
+		"hint":  0,
+		"info":  1,
+		"warn":  2,
+		"error": 3,
+	}
+
+	verbosity, _ := cmd.Flags().GetInt("verbose")
+	logbasics.Initialize(log.LstdFlags, verbosity)
+
+	inputFilename, err := cmd.Flags().GetString("state")
+	if err != nil {
+		return fmt.Errorf("failed getting cli argument 'state'; %w", err)
+	}
+
+	minimumSeverity, err := cmd.Flags().GetString("fail-severity")
+	if err != nil {
+		return fmt.Errorf("failed getting cli argument 'fail-severity'; %w", err)
+	}
+
+	onlyFailures, err := cmd.Flags().GetBool("display-only-failures")
+	if err != nil {
+		return fmt.Errorf("failed getting cli argument 'display-only-failures'; %w", err)
+	}
+
+	var outputFormat string
+	{
+		outputFormat, err = cmd.Flags().GetString("format")
+		if err != nil {
+			return fmt.Errorf("failed getting cli argument 'format'; %w", err)
+		}
+		outputFormat = strings.ToUpper(outputFormat)
+	}
+
+	inputRuleset, err := cmd.Flags().GetString("ruleset")
+
+	if err != nil || inputRuleset == "" {
+		return fmt.Errorf("failed getting cli argument 'ruleset';")
```

Comment:
Wouldn't it be better to differentiate the two cases? e.g. signal `--ruleset` as required if missing and fail with a specific error in case of error while parsing the value.

Note,  all these `cmd.Flags().Get*` can be replaced by the appropriate `StringVarP` / `BoolVarP` etc which automatically parse the values.